src/boot/bootctl.c

```diff
@@ -2331,29 +2331,28 @@ static int verb_is_installed(int argc, char *argv[], void *userdata) {
 
 static int parse_timeout(const char *arg1, char16_t **ret_timeout, size_t *ret_timeout_size) {
         char utf8[DECIMAL_STR_MAX(usec_t)];
-        char16_t *encoded;
-        usec_t timeout;
         int r;
 
         assert(arg1);
         assert(ret_timeout);
         assert(ret_timeout_size);
 
-        if (streq(arg1, "menu-force"))
-                timeout = USEC_INFINITY;
-        else if (streq(arg1, "menu-hidden"))
-                timeout = 0;
+        if (STR_IN_SET(arg1, "menu-force", "menu-hidden"))
+                strcpy(utf8, arg1);
         else {
+                usec_t timeout;
                 r = parse_time(arg1, &timeout, USEC_PER_SEC);
                 if (r < 0)
                         return log_error_errno(r, "Failed to parse timeout '%s': %m", arg1);
-                if (timeout != USEC_INFINITY && timeout > UINT32_MAX * USEC_PER_SEC)
+                if (timeout != USEC_INFINITY && timeout > UINT32_MAX * USEC_PER_SEC) {
                         log_warning("Timeout is too long and will be treated as 'menu-force' instead.");
-        }
+                        timeout = USEC_INFINITY;
+                }
 
-        xsprintf(utf8, USEC_FMT, MIN(timeout / USEC_PER_SEC, UINT32_MAX));
+                xsprintf(utf8, USEC_FMT, MIN(timeout / USEC_PER_SEC, UINT32_MAX));
+        }
```

Comment:
so, what about this: instead of having only utf8[] as an array, and then copying stuff into it that might be constant. Just rename utf8[] to buf[] and make utf8 a `const char*` pointer, that either points to buf[] (in ase of numeric value) or points to the literal string "menu-force" and "menu-hidden". 

That would mean: no ugly strcpy(), and everything is clean an properly sized.